CHANGELOG.md

```diff
@@ -4,6 +4,26 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.4.3] - TBD
+
+### Added
+- Sharded Grad Scaler works with cpu offload in mixed and full precision.
+- Chore: `Pre-Commit` will be the de-facto standard to run all linter checks (black, isort, mypy, flake8 etc.)
+on all commits locally and in CI. [#849]
+
+### Changed
+- Cleanup: Moving forward we would be testing all of our code with Python 3.9.7, CUDA 11.2 and the following three versions of PyTorch [#847]:
+  - the most recent stable version
+  - the most recent LTS version
+  - a recent nightly build
+
+- Cleanup: Developer requirements of python packages will no longer be split into `requirements-dev.txt` and
+`requirements-test.txt`. All package requirements have been merged into `requirements-dev.txt`. [#853]
+
+### Removed
+- Cleanup: `meta.yml` in conda/recipe is no longer supported. [#853]
+- Cleanup: `requirements-test.txt` will no longer be needed as all required libraries are listed in `requirements-dev.txt`. [#853]
```

Comment:
We don't need multiple CHANGELOG entries for the same PR. You can have a single entry under Added. We don't have entries like Changed or Removed. We just have Fixed and Added and I prefer just maintaining two sections.

Reply:
These cleanup changes are minor and may not even be worth mentioning, in some respect. On the other hand, I think we should have sections like Changed/Removed. How else would we inform our users if we deprecate/remove an API in the future? Also I'm following the conventions laid out in https://keepachangelog.com/en/1.0.0/

Reply:
I think keeping the sections short helps users understand what is going on with FairScale. I personally prefer only 1 comment per PR. If you end up with multiple comments that actually tells me you should be sending multiple PRs and are doing too much in 1. We can always change conventions so I prefer not adding too many sections :)  I think Changed/Removed are not very different and folks are going to have different interpretations. I agree with Deprecated. Maybe we can add that if we deprecate an API since we need to call that out.